include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -115,38 +115,37 @@ namespace VectorTools
     // vectorfunction multiple times at the
     // same point on a cell.
     //
-    // note that we have to set up all of the
-    // following arrays for each of the
-    // elements in the FECollection (which
-    // means only once if this is for a regular
-    // DoFHandler)
-    std::vector<std::vector<Point<dim> > > unit_support_points (fe.size());
+    // First check that the desired components are interpolating.
     for (unsigned int fe_index=0; fe_index<fe.size(); ++fe_index)
       {
-        unit_support_points[fe_index] = fe[fe_index].get_unit_support_points();
-        Assert ((unit_support_points[fe_index].size() != 0)||(fe[fe_index].dofs_per_cell == 0),
-                ExcNonInterpolatingFE());
-      }
+        for (unsigned int component_index = 0; component_index < n_components; ++component_index)
+          {
+            if (component_mask[component_index] == true)
+              {
+                std::pair<unsigned int, unsigned int> base_index = fe[fe_index].component_to_base_index(component_index);
 
+                AssertThrow ((fe[fe_index].base_element(base_index.first).has_support_points()) ||
```

Comment:
That's not how we use `Assert` and `AssertThrow`, though. `Assert` is used to catch things that are clearly programming bugs, which in the current case would be that you try to interpolate a non-interpolatory element. Furthermore, you'll get that same error every time you run it, so a static assertion is all that's required. This is what `Assert` does.

We use `AssertThrow` to catch things that may differ from one run to the next, such as a file not existing or a directory being not writable, or an input parameter with an invalid value.

So, here I would suggest using `Assert`. You can't easily test that you get the correct error this way, but that's ok. You can at least test in debug mode that you get the error if you used the `deal_II_exceptions::disable_abort_on_exception` function, though. grep through the existing tests to see how it is used.